src/uct/tcp/tcp_ep.c

```diff
@@ -135,7 +136,8 @@ static void uct_tcp_ep_cleanup(uct_tcp_ep_t *ep)
         uct_tcp_ep_ctx_reset(&ep->rx);
     }
 
-    if (ep->events && (ep->fd != -1)) {
+    if ((ep->events != 0) && (ep->fd != -1)) {
+        ucs_assert(!(ep->flags & UCT_TCP_EP_FLAG_FAILED));
```

Comment:
maybe it's failed but fd still not closed
i'd remove ep->fd!=-1 check here

Reply:
done